massa-async-pool/src/lib.rs

```diff
@@ -16,33 +16,32 @@
 //!
 //! ```json
 //! {
-//!     "sender": "xxxx",  // address that sent the message and spent max_gas*gas_price+coins on emission
+//!     "sender": "xxxx",  // address that sent the message and spent max_gas+coins on emission
 //!     "slot": {"period": 123455, "thread": 11},  // slot at which the message was emitted
 //!     "emission_index": 212,  // index of the message emitted in this slot
 //!     "destination": "xxxx",  // target address
 //!     "handler": "handle_message",  // name of the handler function to call in the target SC
 //!     "validity_start": {"period": 123456, "thread": 12},  // the message can be handled starting from the validity_start slot (included)
 //!     "validity_end": {"period": 123457, "thread": 16},  // the message can be handled until the validity_end slot (excluded)
 //!     "max_gas": 12334,  // max gas available when the handler is called
-//!     "gas_price": "124.23",  // gas price for the handler call
 //!     "coins": "1111.11",  // amount of coins to transfer to the destination address when calling its handler
 //!     "data": { ... any object ... }  // data payload of the message, passed as the sole parameter of the destination handler when called
 //! }
 //! ```
 //!
 //! ## How to send a message during bytecode execution
 //!
-//! * messages are sent using an ABI: `send_message(target_address, target_handler, validity_start, validity_end, max_gas, gas_price, coins, data: JSON string) -> Result<(), ABIReturnError>`. Note that data has a configuration defined `max_async_message_data_size`.
+//! * messages are sent using an ABI: `send_message(target_address, target_handler, validity_start, validity_end, max_gas, coins, data: JSON string) -> Result<(), ABIReturnError>`. Note that data has a configuration defined `max_async_message_data_size`.
 //! * when called, this ABI does this:
 //!   * it consumes `compute_gas_cost_of_message_storage(context.current_slot, validity_end_slot)` of gas in the current execution. This allows making the message emission more gas-consuming when it requires storing the message in queue for longer
-//!   * it consumes `max_gas * gas_price + coins` coins from the sender
+//!   * it consumes `max_gas + coins` coins from the sender
 //!   * it generates an `AsyncMessage` and stores it in an asynchronous pool
 //!
-//! Note that `max_gas*gas_price` coins are burned when sending the message.
+//! Note that `max_gas` coins are burned when sending the message.
```

Comment:
Gas and coins are fundamentally different resources.

`fee` coins are burned when sending the message